request/request.go

```diff
@@ -175,6 +175,11 @@ func Size(url, refer string) (int64, error) {
 		return 0, err
 	}
 	s := h.Get("Content-Length")
+	if s == "" {
+		var err error
+		err = fmt.Errorf("Content-Length is not present")
+		return 0, err
```

Comment:
return `0, errors.New("Content-Length is not present")` would be better

Reply:
absolutely, yeah!